providers/implementations/ciphers/cipher_chacha20_hw.c

```diff
@@ -102,6 +92,11 @@ static int chacha20_cipher(PROV_CIPHER_CTX *bctx, unsigned char *out,
         memset(ctx->buf, 0, sizeof(ctx->buf));
         ChaCha20_ctr32(ctx->buf, ctx->buf, CHACHA_BLK_SIZE,
                        ctx->key.d, ctx->counter);
+
+        /* propagate counter overflow */
+        if (!++ctx->counter[0])
+            ctx->counter[1]++;
+            
```

Comment:
the CI picked up some EOL whitespace here (empty / blank line is fine, but no spaces / tabs; probably your editor is jerking you around)

Reply:
Also style issue - please use `++ctx->counter[0] == 0` instead of a negation.